src/smt/sygus_solver.cpp

```diff
@@ -279,100 +325,39 @@ void SygusSolver::checkSynthSolution(Assertions& as)
-    {
-      Trace("check-synth-sol")
-          << "  " << pair.first << " --> " << pair.second << "\n";
-      fvars.push_back(pair.first);
-      fsols.push_back(pair.second);
-    }
+    Trace("check-synth-sol")
+        << "  " << pair.first << " --> " << pair.second << "\n";
+    fvars.push_back(pair.first);
+    fsols.push_back(pair.second);
   }
+
   Trace("check-synth-sol") << "Starting new SMT Engine\n";
 
   Trace("check-synth-sol") << "Retrieving assertions\n";
   // Build conjecture from original assertions
-  const context::CDList<Node>& alist = as.getAssertionList();
-  Assert(options().smt.produceAssertions)
-      << "Expected produce assertions to be true when checking synthesis "
-         "solution";
-  // auxiliary assertions
-  std::vector<Node> auxAssertions;
-  // expand definitions cache
-  std::unordered_map<Node, Node> cache;
-  for (const Node& assertion : alist)
-  {
-    if (isVerboseOn(1))
-    {
-      verbose(1) << "SyGuS::checkSynthSolution: checking assertion "
-                 << assertion << std::endl;
-    }
-    Trace("check-synth-sol") << "Retrieving assertion " << assertion << "\n";
-    // Apply any define-funs from the problem.
-    Node n = d_smtSolver.getPreprocessor()->expandDefinitions(assertion, cache);
-    if (isVerboseOn(1))
-    {
-      verbose(1) << "SyGuS::checkSynthSolution: -- expands to " << n
-                 << std::endl;
-    }
-    Trace("check-synth-sol") << "Expanded assertion " << n << "\n";
-    if (conjs.find(n) == conjs.end())
-    {
-      Trace("check-synth-sol") << "It is an auxiliary assertion\n";
-      auxAssertions.push_back(n);
-    }
-    else
-    {
-      Trace("check-synth-sol") << "It is a synthesis conjecture\n";
-    }
-  }
   // check all conjectures
-  for (Node conj : conjs)
+  for (const Node& conj : conjs)
   {
     // Start new SMT engine to check solutions
     std::unique_ptr<SolverEngine> solChecker;
-    initializeSubsolver(solChecker, d_env);
+    initializeSygusSubsolver(solChecker, as);
     solChecker->getOptions().smt.checkSynthSol = false;
     solChecker->getOptions().quantifiers.sygusRecFun = false;
-    // get the solution for this conjecture
-    std::vector<Node>& fvars = fvarMap[conj];
-    std::vector<Node>& fsols = fsolMap[conj];
+    Assert(conj.getKind() == FORALL);
+    Node conjBody = conj[1];
+    // we must expand definitions here, since define-fun may contain the
+    // function-to-synthesize, which needs to be substituted.
+    conjBody = d_smtSolver.getPreprocessor()->expandDefinitions(conjBody);
     // Apply solution map to conjecture body
-    Node conjBody;
-    /* Whether property is quantifier free */
-    if (conj[1].getKind() != EXISTS)
-    {
-      conjBody = conj[1].substitute(
-          fvars.begin(), fvars.end(), fsols.begin(), fsols.end());
-    }
-    else
-    {
-      conjBody = conj[1][1].substitute(
-          fvars.begin(), fvars.end(), fsols.begin(), fsols.end());
-
-      /* Skolemize property */
```

Comment:
why is this no longer necessary?

Reply:
It was never necessary to begin with, the subsolver can skolemize itself.